pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/periodictask/ControllerPeriodicTask.java

```diff
@@ -54,14 +54,22 @@ public ControllerPeriodicTask(String taskName, long runFrequencyInSeconds, long
     _controllerMetrics = controllerMetrics;
   }
 
+  // Determine if this task can run on the specified table. Task can run on all tables for which the controller is lead
+  // if "tablename" property is not set. However, if "tablename" property is set (by calling the /periodictask/run
+  // controller API), then the task will only run on the specified by the "tablename" property key.
+  private boolean runTaskForTable(String tableNameWithType) {
+    return _leadControllerManager.isLeaderForTable(tableNameWithType) && (_activePeriodicTaskProperties == null
+        || ((String) _activePeriodicTaskProperties.get("tablename")).equalsIgnoreCase(tableNameWithType));
```

Comment:
```suggestion
        || ((String) _activePeriodicTaskProperties.get("tablename")).equals(tableNameWithType));
```

Table `FooBar` is different from table `fooBar`

Reply:
Done